pkg/internal/credentials.go

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *
+ */
+
+package internal
```

Comment:
could we change the name of this package? "internal" doesn't say much about the package and I am afraid that many weird things could land here in the future.

Reply:
`internal` is a convention in Go, see e.g. https://stackoverflow.com/questions/33351387/how-to-use-internal-packages. The package name should stay as it is, the implication is that any exported stuff from this package and its subpackages is only visible in the current project.

Reply:
I know `internal` convention in go, I don't see any reason why we should use `internal` in this project... it's not being consumed by other projects. It's not a library.

Reply:
..and in this case you have credentials directly under `internal` pkg, that's why it doesn't say much. However if you really want to keep it like that then I won't block.

Reply:
I don't have a strong preference. I am not very fond of this `internal` stuff either, but it's already used in other extension providers, e.g. for AWS or Azure, so I'd say we could keep it as it is. Regarding "it's not being consumed by other projects" - you never know. The idea is that anything in an `internal` package is not considered part of an API, and therefore doesn't need to be maintained compatible. 
There is also a testability aspect - previously the Gardener team was fond of `internal` packages because they allowed them to create many testable functions that were called by more complex logic, and then add tests for them, while avoiding testing the more complex logic. I am strongly in favor of a different approach with respect to testing, (mocks and dependency injection) and this gradually came to be preferred, however `internal` packages can still be found in a lot of places.

Reply:
>  Regarding "it's not being consumed by other projects" - you never know

well, that should not be, for sure that would not be our problem, as I said before this is not a library, any particular function can change its name, disappear or change the way how it works at any point of time (even within a prepared release, e.g. backports). We care only about the extension that must be working correctly and we don't provide any backward API compatibility (except k8s API compatibility).


> I don't have a strong preference. I am not very fond of this internal stuff either, but it's already used in other extension providers, e.g. for AWS or Azure, so I'd say we could keep it as it is.

Ok, I will not change style how you write providers. Resolving conversation.